SIGSEGV crash due to bad pointer dereference.

First observed on bom02 due to a very high container restart rate. Some evidence on other nodes but at much less frequency. This could be the result of client populations.

```
98.160.171.234 - - [19/Jul/2019:07:56:21 +0000] "GET /ndt_protocol HTTP/1.1" 200 0
2019/07/19 07:56:31 ndt5.go:118: Handling connection 63.243.240.101:3010<=WS(S),JSON=>98.160.171.234:47570
2019/07/19 07:56:34 server.go:3012: http: TLS handshake error from [2600:3c02::17:d804]:47730: EOF
2019/07/19 07:56:37 web100_linux.go:24: Summarized data: &{0 14 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0}
2019/07/19 07:56:39 protocol.go:274: Connection is not a TCPConn
2019/07/19 07:56:39 ndt5.go:64: Wrote /var/spool/ndt/ndt5/2019/07/19/ndt-vdzbg_1563218806_00000000000044FE.json
2019/07/19 07:56:39 ndt5.go:104: Test failed, but we recovered: runtime error: invalid memory address or nil pointer dereference
2019/07/19 07:56:39 close.go:31: Could not close connection (close tcp 63.243.240.101:3001->195.206.104.52:35898: use of closed network connection)
2019/07/19 07:56:41 c2s.go:99: Ended C2S test on 63.243.240.101:43435<=WS(S),JSON=>98.160.171.234:46416
2019/07/19 07:56:41 c2s.go:119: 63.243.240.101:3010<=WS(S),JSON=>98.160.171.234:47570 sent us 5242.633287687231 Kbps
2019/07/19 07:56:41 ndt5.go:216: You uploaded at 5242.6333 and downloaded at 0.0000
2019/07/19 07:56:41 ndt5.go:64: Wrote /var/spool/ndt/ndt5/2019/07/19/ndt-vdzbg_1563218806_0000000000004507.json
2019/07/19 07:56:41 close.go:31: Could not close connection (tls: use of closed connection)
98.160.171.234 - - [19/Jul/2019:07:56:31 +0000] "GET /ndt_protocol HTTP/1.1" 200 0
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x82d8f2]

goroutine 12245 [running]:
github.com/m-lab/ndt-server/ndt5/protocol.(*netConnection).Close(0xc0001b2550, 0x9701f8, 0xd830a0)
        <autogenerated>:1 +0x32
github.com/m-lab/go/warnonerror.Close(0x7f5759bbdfb8, 0xc0001b2550, 0x95af09, 0x1f, 0xc0001b2550, 0x45ccc6)
        /go/src/github.com/m-lab/go/warnonerror/close.go:29 +0x35
github.com/m-lab/ndt-server/ndt5/c2s.ManageTest.func2.1(0x7f5759bbdfd8, 0xc0001b2550)
        /go/src/github.com/m-lab/ndt-server/ndt5/c2s/c2s.go:80 +0x7c
created by github.com/m-lab/ndt-server/ndt5/c2s.ManageTest.func2
        /go/src/github.com/m-lab/ndt-server/ndt5/c2s/c2s.go:78 +0x49
```

--- a/ndt5/singleserving/server.go
+++ b/ndt5/singleserving/server.go
@@ -187,6 +187,9 @@ func (ps *plainServer) ServeOnce(ctx context.Context) (protocol.MeasuredConnecti
 	if err != nil {
 		return nil, err
 	}
+	if conn == nil {
+		return nil, errors.New("nil conn, nil err: " + derivedCtx.Err().Error())
+	}
 	return protocol.AdaptNetConn(conn, conn), nil
 }
 

